public/app/plugins/panel/gauge/GaugePanel.tsx

```diff
@@ -15,6 +16,13 @@ export class GaugePanel extends PureComponent<Props> {
       nullValueMode: NullValueMode.Ignore,
     });
 
-    return <Gauge timeSeries={vmSeries} {...this.props.options} width={width} height={height} />;
+    return (
+      <Gauge
+        timeSeries={vmSeries}
+        {...this.props.options}
+        width={width - 2 * variables.panelHorizontalPadding}
+        height={height - variables.panelVerticalPadding}
```

Comment:
I wonder if we should do this in PanelChrome instead, then panel plugins do not need to bother:
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/dashgrid/PanelChrome.tsx#L126

Reply:
Well, you are right @torkelo. No matter what viz we render, the size passed shouldn't be wider than panel's content area.